### PR TITLE
feat: adds welcome page & pull / build buttons

### PR DESCRIPTION
**Pull request by cdrage**
feat: adds welcome page & pull / build buttons

### What does this PR do?

* Updates the welcome / empty screen page so there is no more empty
  descriptions
* Adds a build / pull button for a test iamge
* Added API for opening an eternal link
* If pulling for more than 5 seconds, show a warning

### Screenshot / video of UI

https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/ac51a157-f896-498f-80e3-316e22755517

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/150

### How to test this PR?

Go to the welcome page with no builds, click on the pull / build
buttons.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
@slemeur @deboer-tim 

Very simple welcome page that incorporates all that was talked about this week.

Let me know with regards to the design what to improve and then I can add / modify the tests.

TODOs after this PR:
 - On clicking "Build" pre-select the image, this feature will most likely be in following #235

**Comment by slemeur** [comment]
This is very good!

File: packages/frontend/src/lib/BootcEmptyScreen.svelte [hunk 1]
```diff
@@ -1,9 +1,109 @@
...
-  title="No bootable container builds found"
-  message="Start your first build by clicking the 'Build' button" />
+<div class="flex w-full h-full justify-center items-center">
+  <div class="flex flex-col h-full items-center text-center space-y-3">
+    <!-- Bootable Container Icon -->
+    <div class="text-gray-700 py-2">
+      <svelte:component this="{BootcSelkie}" size="120" />
+    </div>
+
+    <h1 class="text-xl pb-4">Welcome to the Bootable Containers image builder!</h1>
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcEmptyScreen.svelte:73]: We're going to have more discussion on which terms to use after this PR, but for now I'd prefer to drop 'the' and 'image builder', i.e. "Welcome to Bootable Containers" and be the intro to bootable containers and not limit ourselves to only building images.

File: packages/frontend/src/lib/BootcEmptyScreen.svelte [hunk 2]
```diff
@@ -1,9 +1,109 @@
...
+      Bootable Containers builds an entire bootable OS from your container image. Utilizing the technology of a
+      <Link externalRef="{centosBootcSite}">compatible image</Link>, <Link externalRef="{bootcImageBuilderSite}"
+        >bootc-image-builder</Link
+      >, and <Link externalRef="{bootcSite}">bootc</Link>, your container image is transformed into a bootable disk
+      image.
+    </p>
+
+    <p class="text-gray-700 pb-1 max-w-xl">
+      Create your first disk image by {imageExists ? 'building' : 'pulling'} the <Link
+        externalRef="{`https://${exampleImage}`}">example image</Link
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcEmptyScreen.svelte:85]: "example container image"? Feels like we should keep being really clear about container images vs disk images.

File: packages/frontend/src/lib/BootcEmptyScreen.svelte [hunk 3]
```diff
@@ -1,9 +1,109 @@
...
+        inProgress="{pullInProgress}"
+        title="Pull image">Pull {exampleImage}</Button>
+    {/if}
+    {#if displayDisclaimer}
+      <p class="text-amber-500 text-xs">The file size of the image is over 1.5GB and may take a while to download.</p>
+    {/if}
+
+    <p class="text-gray-700 pt-8 max-w-xl">
+      Want to learn more including building your own Containerfile? Check out the <Link externalRef="{extensionSite}"
+        >extension documentation</Link
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcEmptyScreen.svelte:105]: I think you can drop 'external', most people would assume docs are on the web.

File: packages/frontend/src/lib/upstream/Link.svelte [hunk 4]
```diff
@@ -23,8 +24,7 @@ function click() {
   if (internalRef) {
     router.goto(internalRef);
   } else if (externalRef) {
-    // TODO: Does not work at the moment
```
Comment (deboer-tim) [inline packages/frontend/src/lib/upstream/Link.svelte:26]: Can't access PD API from within the frontend, I think this is expected.

**Review by deboer-tim — CHANGES_REQUESTED** [review]
Looking really good. Needs to be rebased and I suggested a few tweaks in wording, but if those were done I'd rather merge and have follow-up PRs for things like keeping some form of this open (vs more commits here).

File: packages/frontend/src/lib/upstream/Link.svelte (hunk 4 above)
Comment (cdrage) [reply packages/frontend/src/lib/upstream/Link.svelte:26]: Have added this as a new API command / fixed the "upstream" code. We had to add the API call for externalLink for it to work: https://github.com/containers/podman-desktop-extension-bootc/pull/241/files#diff-db53f7a2ddea88bd5039df3b77b7bf1141e88c3c05858eb035f5ce05a1c1e35bR31

**Comment by cdrage** [comment]
Updated to also now preselect the image when going to the build page:


https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/8c9f6c89-79ad-4206-82fb-9b0ba5ed435b

File: packages/frontend/src/lib/BootcEmptyScreen.svelte (hunk 3 above)
Comment (deboer-tim) [reply packages/frontend/src/lib/BootcEmptyScreen.svelte:110]: Sorry, I meant the word 'external' :)

**Review by deboer-tim — APPROVED** [review]

**Pull request merged**